TSDB: Make time boundaries settings required in tsdb indices

weizijun opened the pull request:
Make `index.time_series.start_time` and `index.time_series.end_time` settings as required settings in tsdb indices.

It will change many tests, among which a time_series index is created, and it will add the `index.time_series.start_time` and `index.time_series.end_time` settings

weizijun committed:
temp

weizijun committed:
Merge branch 'upstream/master' into tsdb-required-range-settings

* upstream/master: (29 commits)
  Fix typo (#80925)
  Increase docker compose timeouts for CI builds
  TSDB: fix error without feature flag (#80945)
  [DOCS] Relocate `index.mapping.dimension_fields.limit` setting docs (#80964)
  Explicit write methods for always-missing values (#80958)
  TSDB: move TimeSeriesModeIT to yaml tests (#80933)
  [ML] Removing temporary debug (#80956)
  Remove unused ConnectTransportException#node (#80944)
  Reinterpret dots in field names as object structure (#79922)
  Remove obsolete typed legacy index templates (#80937)
  Remove unnecessary shuffle in unassigned shards allocation. (#65172)
  TSDB: Tests for nanosecond timeprecision timestamp just beyond the limit (#80932)
  Cleanup SLM History Item .equals (#80938)
  Rework breaking changes for new structure (#80907)
  [DOCS] Fix elasticsearch-reset-password typo (#80919)
  [ML] No need to use parent task client when internal infer delegates (#80905)
  Fix shadowed vars pt6 (#80899)
  add ignore info (#80924)
  Fix several potential circuit breaker leaks in Aggregators (#79676)
  Extract more standard metadata from binary files (#78754)
  ...

weizijun committed:
fixup

weizijun committed:
fixup

weizijun committed:
fixup

weizijun committed:
Merge branch 'upstream/master' into tsdb-required-range-settings

* upstream/master: (55 commits)
  Fix ComposableIndexTemplate equals when composed_of is null (#80864)
  Optimize DLS bitset building for matchAll query (#81030)
  URL option for BaseRunAsSuperuserCommand (#81025)
  Less Verbose Serialization of Snapshot Failure in SLM Metadata (#80942)
  Fix shadowed vars pt7 (#80996)
  Fail shards early when we can detect a type missmatch (#79869)
  Delegate Ref Counting to ByteBuf in Netty Transport (#81096)
  Clarify `unassigned.reason` docs (#81017)
  Strip blocks from settings for reindex targets (#80887)
  Split off the values supplier for ScriptDocValues (#80635)
  [ML] Switch message and detail for model snapshot deprecations (#81108)
  [DOCS] Update xrefs for snapshot restore docs (#81023)
  [ML] Updates visiblity of validate API (#81061)
  Track histogram of transport handling times (#80581)
  [ML] Fix datafeed preview with remote indices (#81099)
  [ML] Fix acceptable model snapshot versions in ML deprecation checker (#81060)
  [ML] Add logging for failing PyTorch test (#81044)
  Extending the timeout waiting for snapshot to be ready (#81018)
  [ML] Fix incorrect logging of unexpected model size error (#81089)
  [ML] Make inference timeout test more reliable (#81094)
  ...

weizijun committed:
update tests

weizijun commented:
@elasticsearchmachine  update branch

elasticmachine commented:
Pinging @elastic/es-analytics-geo (Team:Analytics)

nik9000 reviewed (commented):
I think it'd be nice to  use times in the tests that are a little closer to the test data. If we're really make truly random times from 1 to Long.MAX_VALUE it's cool to have those as bounds, but for the tests I'd feel better with bounds that are a day or a month long or something like that.

nik9000 commented:
@elasticmachine, test this please

weizijun committed:
update start_time and end_time in tests

weizijun committed:
Merge branch 'upstream/master' into tsdb-required-range-settings

* upstream/master:
  [ML] Parent datafeed actions to the datafeed's persistent task (#81143)
  Simplify ConcreteIndices and its usage in TransportBulkAction (#81098)
  Unmute DataStreamsSnapshotsIT#testRestoreDataStreamAliasWithConflictingIndicesAlias() test (#81142)
  TSDB: Do not allow index splits for time series indices (#81125)
  Reduce verbosity-increase timeout to 3m (#81118)
  Mute DataStreamsSnapshotsIT#testRestoreDataStreamAliasWithConflictingIndicesAlias() test
  Fix stopping of old elasticsearch cluster (#81059)
  Fix data stream alias validation. (#81040)
  Add replicated field to get data stream api response. (#80988)

nik9000 commented on rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/60_add_dimensions.yml:
```diff
@@ -114,8 +114,8 @@ add dimensions to no dims with dynamic_template over bulk:
-                  start_time: 1
-                  end_time: 253402300799998
+                  start_time: 2021-04-28T00:00:00Z
+                  end_time: 2999-04-28T00:00:00Z
```
Maybe use something closer to the range of data we have?

weizijun replied:
done, and there left some failed tests, I will run gradle check to find all failed tests, and fix it tomorrow

weizijun committed:
update end_time in tests

weizijun commented:
@nik9000 can I change all tsdb yaml test's skip config from:
```
      version: " - 7.99.99"
      reason: introduced in 8.0.0
```
to
```
      version: " - 8.0.99"
      reason: introduced in 8.1.0
```
to pass the mixed-cluster tests

nik9000 commented:
> to pass the mixed-cluster tests

:+1:

nik9000 commented:
@elasticmachine, test this please

weizijun committed:
fixup

weizijun commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into tsdb-required-range-settings

nik9000 commented:
run elasticsearch-ci/release-tests

nik9000 commented:
> @nik9000 the failed elasticsearch-ci/release-tests seems that the problem of the environment

:+1:. Let's see if it's better a second time.

nik9000 commented:
run elasticsearch-ci/release-tests

nik9000 commented:
@elasticmachine, update branch
